lib/active_merchant/billing/gateways/adyen.rb

```diff
@@ -97,7 +98,15 @@ def add_extra_data(post, options)
         post[:selectedBrand] = options[:selected_brand] if options[:selected_brand]
         post[:deliveryDate] = options[:delivery_date] if options[:delivery_date]
         post[:merchantOrderReference] = options[:merchant_order_reference] if options[:merchant_order_reference]
-        post[:shopperInteraction] = options[:shopper_interaction] if options[:shopper_interaction]
+      end
+
+      def add_shopper_interaction(post, payment, options={})
+        if payment.verification_value
```

Comment:
This is short enough that you _could_ do `shopper_interaction = payment.verification_value ? 'Ecommerce' : 'ContAuth'`, but that's 100% your call.